pupil_src/shared_modules/player_methods.py

```diff
@@ -58,6 +58,21 @@ def __init__(self, data=(), data_ts=()):
             self.data_ts = self.data_ts[self.sorted_idc]
             self.data = self.data[self.sorted_idc].tolist()
 
+    def by_ts(self, ts):
+        '''
+        :param ts: matching or closest timestamp to extract.
+        :return: datum that is matching or None if none is found
+        '''
+        found_i = np.searchsorted(self.data_ts, ts)
+        # searchsorted returns len(array) if the query is bigger than all elements
+        in_range = found_i < len(self.data_ts)
```

Comment:
I would not check like this. Rather catch the exception. better even leave it unhandles. This is something the caller should handle.

Reply:
Actually, I disagree that the caller should be handling this. Yes, this check should not be performed explicitly but rather through a exception -- but it should be handled within the method. Either the `ts` is present in `data_ts` and the datum is returned, or it is not and `None` is returned. There is no reason why this function should throw an exception if `ts` is bigger than the last element in `data_ts`

Reply:
Thank you, I was not aware of this. It now throws a ValueError

Reply:
For completion: A `ValueError` is raised instead of returning `None`. This function therefore returns a valid result or raises a `ValueError`.